src/Controller/Badge/ComposerLockController.php

```diff
@@ -49,6 +49,9 @@ public function composerLock(
         $badge = $composerLockBadge->createComposerLockBadge($repository, $format);
         $image = $imageFactory->createFromBadge($badge);
 
-        return ResponseFactory::createFromImage($image, 200);
+        $maxage = 60 * 60;
+        $smaxage = 24 * 60 * 60;
```

Comment:
This way you are serving stale contents to the clients ... authorising the shared caches (your CDN, a caching gateway in the middle between the server and "a client", an company proxy, an ISP accelerator, etc ..) to keep an expired object and serve it to the client ...

This configuration is quite common in the net, the reason is ppl tries to lower the load on the poor man attic-stored "server", but it is wrong.

Just remove s-maxage, and it would be ok.  If you want to instruct a shared cache to do it's job, set it lower than max age.

My 2 cents ;)

Reply:
If I'm not wrong, maxage is handled by the browser, and smaxage by the proxy, I'd like to keep a longer copy in the shared cache rather than the browser's one.

Also, on https://symfony.com/doc/current/the-fast-track/en/21-cache.html#adding-http-cache-headers I've got the same idea...

Reply:
As said .. if you permit your "shared cacheS" (also the ones you can't control, mind this fact) to keep a stale, expired, copy .. then you are serving old contents to your clients.

Reply:
fixed